io/json/include/detray/io/json_io.hpp

```diff
@@ -0,0 +1,75 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// This header is used to disable GCC errors that could be occuring in
+// nlohmann/json.hpp
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wimplicit-fallthrough"
+#include <nlohmann/json.hpp>
+#pragma GCC diagnostic pop
+
+namespace ALGEBRA_TRANSFORM3_NAMESPACE {
+
+void to_json(nlohmann::json& j, const __plugin::transform3<detray::scalar>& t) {
```

Comment:
As a reminder, one of the big rewrites that is still ahead of us is to get rid of these preprocessor tricks from the code. I just didn't psych myself up sufficiently yet to spend the needed couple of days on it.

In this PR it's fine to continue using this pattern I guess. But I intend to get rid of this design pattern completely at one point.